src/main/java/uk/nhs/hee/tis/trainee/sync/service/UserDesignatedBodySyncService.java

```diff
@@ -51,10 +53,17 @@ public void syncRecord(Record userDesignatedBody) {
       throw new IllegalArgumentException(message);
     }
 
+    String userName = userDesignatedBody.getData().get(USER_DB_USER_NAME);
+    String designatedBodyCode = userDesignatedBody.getData().get(USER_DB_DBC);
+    Optional<UserDesignatedBody> udbOptional =
+        repository.findByUserNameAndDesignatedBodyCode(userName, designatedBodyCode);
+
     if (userDesignatedBody.getOperation().equals(DELETE)) {
-      repository.deleteById(userDesignatedBody.getTisId());
+      udbOptional.ifPresent(designatedBody -> repository.deleteById(designatedBody.getTisId()));
     } else {
-      repository.save((UserDesignatedBody) userDesignatedBody);
+      udbOptional.ifPresentOrElse(
+          db -> log.info("User designated body record {} already exists.", db),
```

Comment:
Since this table only has the two fields, there is no point in deleting and reinserting the record.